src/vs/workbench/contrib/terminalContrib/accessibility/test/browser/terminalAccessibleBuffer.test.ts

```diff
@@ -0,0 +1,116 @@
+import { TerminalConfigHelper } from 'vs/workbench/contrib/terminal/browser/terminalConfigHelper';
+import { getTerminalUri } from 'vs/workbench/contrib/terminal/browser/terminalUri';
+import { XtermTerminal } from 'vs/workbench/contrib/terminal/browser/xterm/xtermTerminal';
+import { ITerminalConfiguration } from 'vs/workbench/contrib/terminal/common/terminal';
+import { AccessibleBufferWidget } from 'vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBuffer';
+import { ILifecycleService } from 'vs/workbench/services/lifecycle/common/lifecycle';
+import { TestLifecycleService } from 'vs/workbench/test/browser/workbenchTestServices';
+import { TestStorageService } from 'vs/workbench/test/common/workbenchTestServices';
+import { Terminal } from 'xterm';
+class TestXtermTerminal extends XtermTerminal {
+
+}
+
+const defaultTerminalConfig: Partial<ITerminalConfiguration> = {
+	fontFamily: 'monospace',
+	fontWeight: 'normal',
+	fontWeightBold: 'normal',
+	gpuAcceleration: 'off',
+	scrollback: 1000,
+	fastScrollSensitivity: 2,
+	mouseWheelScrollSensitivity: 1,
+	unicodeVersion: '6'
+};
+
+suite('Accessible buffer', () => {
+	let instantiationService: TestInstantiationService;
+	let configurationService: TestConfigurationService;
+	let themeService: TestThemeService;
+	let viewDescriptorService: TestViewDescriptorService;
+	let xterm: TestXtermTerminal;
+	let capabilities: TerminalCapabilityStore;
+	let configHelper: TerminalConfigHelper;
+	let terminalInstance: Pick<ITerminalInstance, 'capabilities' | 'onDidRequestFocus' | 'resource'>;
+	let accessibleBufferWidget: AccessibleBufferWidget;
+
+	setup(() => {
+		configurationService = new TestConfigurationService({
+			editor: {
+				fastScrollSensitivity: 2,
+				mouseWheelScrollSensitivity: 1
+			} as Partial<IEditorOptions>,
+			terminal: {
+				integrated: defaultTerminalConfig
+			}
+		});
+		themeService = new TestThemeService();
+		viewDescriptorService = new TestViewDescriptorService();
+
+		instantiationService = new TestInstantiationService();
+		instantiationService.stub(IConfigurationService, configurationService);
+		instantiationService.stub(ILogService, new NullLogService());
+		instantiationService.stub(IStorageService, new TestStorageService());
+		instantiationService.stub(IThemeService, themeService);
+		instantiationService.stub(IAccessibilityService, new TestAccessibilityService());
+		instantiationService.stub(IContextKeyService, new ContextKeyService(configurationService));
+		instantiationService.stub(ILanguageFeaturesService, new LanguageFeaturesService());
+		instantiationService.stub(IViewDescriptorService, viewDescriptorService);
+		instantiationService.stub(IContextMenuService, instantiationService.createInstance(ContextMenuService));
+		instantiationService.stub(ILifecycleService, new TestLifecycleService());
+		instantiationService.stub(ICodeEditorService, new TestCodeEditorService(themeService));
```

Comment:
Not that important now that it's done, but you could avoid all this by creating a class that's just responsible for tracking the cached lines and giving the full buffer. I think it would basically only need the xterm instance in order to do its thing

Reply:
ah yeah I see that. maybe a good refactor to do in the future